doc/languages-frameworks/python.section.md

```diff
@@ -336,21 +336,42 @@ Python versions or modules, which is why they don't go in `python3Packages`.
 
 A distinction is made between applications and libraries, however, sometimes a
 package is used as both. In this case the package is added as a library to
-`python-packages.nix` and as an application to `all-packages.nix`. To reduce
-duplication the `toPythonApplication` can be used to convert a library to an
-application.
+`python-packages.nix` and as an application to the top-level. To reduce
+duplication the `toPythonApplication` function can be used to convert a library
+to an application.
 
-The Nix expression shall use [`buildPythonPackage`](#buildpythonpackage-function) and be called from
-`python-packages.nix`. A reference shall be created from `all-packages.nix` to
-the attribute in `python-packages.nix`, and the `toPythonApplication` shall be
-applied to the reference:
+The Nix expression shall use [`buildPythonPackage`](#buildpythonpackage-function)
+and be called from `python-packages.nix`, as usual. A reference shall be created
+from the top-level to the attribute in `python-packages.nix`, and
+`toPythonApplication` shall be applied to that reference.
+
+At the current moment there are two methods for achieving this. We'll start with
+the legacy one.
+
+##### Legacy Method {#topythonapplication-function-legacy-method}
+
+The legacy method consists in registering a reference from `all-packages.nix`
+to the attribute in `python-packages.nix`, and applying `toPythonApplication` to
+that reference. Let's take `youtube-dl` as an example:
 
 ```nix
 {
   youtube-dl = with python3Packages; toPythonApplication youtube-dl;
 }
 ```
 
+##### New Method {#topythonapplication-function-new-method}
```

Comment:
Feel like it's important to call this out to avoid people trying to migrate things to the new method just because they read it here.

```suggestion
New packages should avoid using the legacy method and prefer the new method.
Existing packages that already use the legacy method shouldn't be migrated to the new method to prevent changing its [override](#sec-pkg-override) interface.

##### New Method {#topythonapplication-function-new-method}
```

Reply:
I added notes. Tell me what do you think.